src/Solver.cpp

```diff
@@ -454,15 +602,34 @@ void Solver::initializeFSRs() {
     /* Assign the Material corresponding to this FSR */
     _FSR_materials[r] = _geometry->findFSRMaterial(r);
 
-    /* Increment number of fissionable FSRs */
-    if (_FSR_materials[r]->isFissionable())
-      _num_fissionable_FSRs++;
-
     log_printf(DEBUG, "FSR ID = %d has Material ID = %d and volume = %f ",
                r, _FSR_materials[r]->getId(), _FSR_volumes[r]);
   }
+}
+
+
+/**
+ * @brief Counts the number of fissionable flat source regions.
+ * @details This routine is used by the Solver::computeEigenvalue(...) 
+ *          routine which uses the number of fissionable FSRs to normalize
+ *          the residual on the fission source distribution.
+ */
+void Solver::countFissionableFSRs() {
+
+  log_printf(INFO, "Counting fissionable FSRs...");
+
+  /* Count the number of fissionable FSRs */
+  std::map<int, Material*> all_materials = _geometry->getAllMaterials();
+  _num_fissionable_FSRs = 0;
+
+  for (int r=0; r < _num_FSRs; r++) {
+    if (_FSR_materials[r]->isFissionable())
+      _num_fissionable_FSRs++;
+  }
 
-  return;
+  if (_num_fissionable_FSRs == 0)
```

Comment:
If `countFissionableFSRs` is called for all `compute...` functions, there will be cases where the `_num_fissionable_FSRs` is 0 (e.g. `point-source.py`). `_num_fissionable_FSRs` being 0 only seems to be a problem if the residual type is `FISSION_SOURCE`. Perhaps you could move this check to the `computeResidual` function inside if the `FISSION_SOURCE` if statement?